src/oapv.c

```diff
@@ -2153,5 +2153,11 @@ int oapvd_info(void *au, int au_size, oapv_au_info_t *aui)
 #endif // ENABLE_DECODER
 ///////////////////////////////////////////////////////////////////////////////
 
-static char *oapv_ver = "0.1.13.1";
-char * oapv_version() { return oapv_ver; }
+char *oapv_version(void)
```

Comment:
Can't this be made a `const char*` instead? With CMake it's trivial to generate the "0.1.13.1" string at configure time, but i assume the issue is with Android?
As is, this looks potentially racy. If it can't be const, then it should allocate a buffer and return it, and the user should then free it.

Also, an int version is still extremely useful for runtime checks, so please add it as a separate function, like `unsigned int oapv_version_int()`

Reply:
To generate the int to be returned? Yes

```
unsigned int oapv_version_int()
{
    return OAPV_VER_SET(OAPV_VER_APISET, OAPV_VER_MAJOR, OAPV_VER_MINOR, OAPV_VER_PATCH);
}
```

Without a function like this, API users that want the version in integer form would need to parse the output of oapv_version()

Reply:
The function may return not just the version number but also an error code, which is a negative value, despite the return type being unsigned, and as such can't be checked at all. For all the API user knows, it just returned a valid version number they can parse with OAPV_VER_GET_*.

Reply:
> The function may return not just the version number but also an error code, which is a negative value, despite the return type being unsigned, and as such can't be checked at all. For all the API user knows, it just returned a valid version number they can parse with OAPV_VER_GET_*.

Again, oapv_version() has changed to support your comments.
Check the PR https://github.com/AcademySoftwareFoundation/openapv/pull/105 .